templates/project.py

```diff
@@ -1,475 +1,530 @@
+    resources.append({
+        'name': 'create-storage-logs-bucket',
+        'type': 'gcp-types/storage-v1:buckets',
+        'properties': {
+            'project': '$(ref.project.projectId)',
+            'name': bucket_name,
+            'lifecycle': {
+                'rule': [
                     {
-                        'consumerId': 'project:' + '$(ref.project.projectId)',
-                        'serviceName': api
+                        'action': {
+                            'type': 'Delete'
+                        },
+                        'condition': {
+                            'age': context.properties.get('storageBucketLifecycle', 180)
+                        }
+
                     }
+                ]
             }
-        )
-
-    # Return the API resources to enable other resources to use them as
-    # dependencies, to ensure that they are created first. For example,
-    # the default VPC or service account.
-    return resources, api_names_list
-
-
-def create_project_iam(dependencies, role_member_list):
-    """ Grant the shared project IAM permissions. """
-
-    policies_to_add = role_member_list
-    resources = [
-        {
-            # Get the IAM policy first, so as not to remove
-            # any existing bindings.
-            'name': 'get-iam-policy',
-            'action': 'gcp-types/cloudresourcemanager-v1:cloudresourcemanager.projects.getIamPolicy', # pylint: disable=line-too-long
-            'properties': {
-                'resource': '$(ref.project.projectId)'
-            },
-            'metadata':
-                {
-                    'dependsOn': dependencies,
-                    'runtimePolicy': ['UPDATE_ALWAYS']
-                }
         },
-        {
-            # Set the IAM policy patching the existing policy
-            # with whatever is currently in the config.
-            'name': 'patch-iam-policy',
-            'action': 'gcp-types/cloudresourcemanager-v1:cloudresourcemanager.projects.setIamPolicy', # pylint: disable=line-too-long
-            'properties':
-                {
-                    'resource': '$(ref.project.projectId)',
-                    'policy': '$(ref.get-iam-policy)',
-                    'gcpIamPolicyPatch':
-                        {
-                            'add': policies_to_add
-                        }
-                }
+        'metadata': {
+            # Only create the bucket once all APIs have been
+            # activated.
+            'dependsOn': api_names_list
         }
-    ]
+    })
+
+    # # Add cloud-storage-analytics@google.com as a writer so it can write logs
+    # # Do it as a separate call so bucket gets default permissions plus this one
+    resources.append({
+        'name': 'add-cloud-storage-writer',
+        'type': 'gcp-types/storage-v1:bucketAccessControls',
+        'properties': {
+            'bucket': bucket_name,
+            'entity': 'group-cloud-storage-analytics@google.com',
```

Comment:
Just curious what the difference is between `group-cloud-storage-analytics@google.com` and `cloud-storage-analytics@google.com`?

Reply:
Huh. I think that's probably supposed to be `group:cloud-storage-analytics@google.com`, but I'll check.

Reply:
I got it from [here](https://cloud.google.com/storage/docs/access-logs#delivery) -- hit JSON API -- but I think it's a typo.